cmd/nerdctl/builder_build_linux_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/containerd/nerdctl/v2/pkg/testutil"
+	"gotest.tools/v3/assert"
+)
+
+func TestBuildContextWithOCILayout(t *testing.T) {
+	testutil.RequiresBuild(t)
+	testutil.RegisterBuildCacheCleanup(t)
+
+	var dockerBuilderArgs []string
+	if testutil.IsDocker() {
+		// Default docker driver does not support OCI exporter.
+		// Reference: https://docs.docker.com/build/exporters/oci-docker/
+		builderName := testutil.SetupDockerContainerBuilder(t)
+		dockerBuilderArgs = []string{"buildx", "--builder", builderName}
+	}
+
+	base := testutil.NewBase(t)
+	imageName := testutil.Identifier(t)
+	t.Cleanup(func() { base.Cmd("rmi", imageName).AssertOK() })
+
+	dockerfile := fmt.Sprintf(`FROM %s
+LABEL layer=oci-layout-parent
+CMD ["echo", "test-nerdctl-build-context-oci-layout-parent"]`, testutil.CommonImage)
+	buildCtx := createBuildContext(t, dockerfile)
+
+	ociLayout := "parent"
+	tarPath := fmt.Sprintf("%s/%s.tar", buildCtx, ociLayout)
+
+	var buildArgs []string
+	if testutil.IsDocker() {
+		buildArgs = dockerBuilderArgs
+	}
+
+	buildArgs = append(buildArgs, "build", buildCtx, fmt.Sprintf("--output=type=oci,dest=%s", tarPath))
+	base.Cmd(buildArgs...).Run()
```

Comment:
`AssertOK()` instead of `Run`?

Reply:
I made some progress here. Client was returning error due to nil manifest list. The workaround I applied was to use image save to create a properly formatted OCI archive. The trade-off was I needed to import distribution/distribution to match against Docker schema2 manifest; however, I think this is fine as the import is small. Alternatively could consider copying the constant if we don't want to take the dependency.